app/controllers/api/v2/infrastructures_controller.rb

```diff
@@ -44,29 +44,26 @@ def profile_curator
              }, status: :not_found) && return
     end
 
-    @profiles = []
-    @apps = BaritoApp.where(status: BaritoApp.statuses[:active])
-
-    @apps.each do |app|
-      @app_group = app.app_group
-      @infrastructure_component = @app_group.infrastructure.infrastructure_components.where(
+    profiles = []
+    AppGroup.all.each do |app_group|
+      es_component = app_group.infrastructure.infrastructure_components.where(
         component_type: 'elasticsearch',
         status: InfrastructureComponent.statuses[:finished],
-      )
-      if @infrastructure_component.blank?
-        next
-      end
+      ).first
+
+      next if es_component == nil
 
-      @infrastructure_component.each do |component|
-        @profiles << {
-          hostname: component.hostname,
-          ipaddress: component.ipaddress,
-          log_retention_days: @app_group.log_retention_days,
-        }
-      end
+      profiles << {
+        ipaddress: es_component.ipaddress,
+        log_retention_days: app_group.log_retention_days,
+        log_retention_days_per_topic: app_group.barito_apps.inject({}) do |app_map, app|
+          app_map[app.topic_name] = app.log_retention_days ? app.log_retention_days : app_group.log_retention_days
```

Comment:
Can you just add topic which has differrent log retention value with app group log retention?